test/framework/easyconfigs/yeb/goolf-1.4.10.yeb

```diff
@@ -28,12 +28,17 @@ toolchain: {name: dummy, version: dummy}
 # compiler toolchain dependencies
 # we need GCC and OpenMPI as explicit dependencies instead of gompi toolchain
 # because of toolchain preperation functions
-dependencies: [
-    *comp,
-    [OpenMPI, 1.6.4, '', *comp],
-    [*blaslib, *blasver, *blas_suff, *comp_mpi_tc],
-    [FFTW, 3.3.3, '', *comp_mpi_tc],
-    [ScaLAPACK, 2.0.2, !join [-, *blas, *blas_suff], *comp_mpi_tc]
-]
+dependencies:
+    - *comp_name: *comp_version
+    - OpenMPI: 1.6.4
+      toolchain: *comp
+    - *blaslib: *blasver
+      versionsuffix: *blas_suff
+      toolchain: *comp_mpi_tc
+    - FFTW: 3.3.3
+      toolchain: *comp_mpi_tc
+    - ScaLAPACK: 2.0.2
+      versionsuffix: !join [-, *blas, *blas_suff]
+      toolchain: *comp_mpi_tc
```

Comment:
this looks a lot clear, especially keeping in mind that it doesn't get nastier than this (specifying both `versionsuffix` and `toolchain` for almost all dependencies is definitely the exception, not the rule)